robotoff/app/core.py

```diff
@@ -165,6 +173,38 @@ def get_predictions(
         return query.iterator()
 
 
+def get_image_predictions(
+    with_logo: Optional[bool] = False,
+    barcode: Optional[str] = None,
+    type: Optional[str] = None,
+    offset: Optional[int] = None,
+    count: bool = False,
+    limit: Optional[int] = 25,
+) -> Iterable[LogoAnnotation]:
+
+    query = LogoAnnotation.select().join(ImagePrediction).join(ImageModel)
```

Comment:
You should not start from LogoAnnotation ? Because you want to return ImagePrediction, not LogoAnnotation !